File: module-1/11_Inheritance/exercise/src/main/java/com/techelevator/SavingsAccount.java
package com.techelevator;

public class SavingsAccount extends BankAccount {

    public SavingsAccount(String accountHolderName, String accountNumber, int balance) {
        super(accountHolderName, accountNumber, balance);
    }

    public SavingsAccount(String accountHolderName, String accountNumber) {
        super(accountHolderName, accountNumber);
    }

    @Override
    public int withdraw(int amountToWithdraw) {
        final int SERVICE_CHARGE = 2;
        if(amountToWithdraw + SERVICE_CHARGE > getBalance()) {
            return getBalance();
        }
        if(getBalance() - amountToWithdraw < 150) {
            return super.withdraw(amountToWithdraw + SERVICE_CHARGE);
        }
        return super.withdraw(amountToWithdraw);
    }


}

File: csharp/module-1/12_Polymorphism/exercise/Exercise/SavingsAccount.cs
namespace Exercise
{
    public class SavingsAccount : BankAccount
    {
        public const decimal LowBalance = 150;
        public const decimal ServiceCharge = 2;

        public SavingsAccount(string accountHolder, string accountNumber) : base(accountHolder, accountNumber)
        {

        }

        public SavingsAccount(string accountHolder, string accountNumber, decimal balance) : base(accountHolder, accountNumber, balance)
        {

        }

        public override decimal Withdraw(decimal amountToWithdraw)
        {
            // Only perform transaction if there's still room for service charge
            if (amountToWithdraw > 0 && (Balance - amountToWithdraw >= ServiceCharge))
            {
                // Withdraw the amount
                base.Withdraw(amountToWithdraw);

                // Assess service charge if balance goes below low balance threshold
                if (Balance < LowBalance)
                {
                    base.Withdraw(ServiceCharge);
                }
            }
            return Balance;
        }
    }
}
